src/neptune/integrations/pandas/__init__.py

```diff
@@ -0,0 +1,102 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import annotations
+
+__all__ = ["to_pandas"]
+
+from datetime import datetime
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Dict,
+    Tuple,
+    Union,
+)
+
+import pandas as pd
+
+from neptune.internal.backends.api_model import (
+    AttributeType,
+    AttributeWithProperties,
+    LeaderboardEntry,
+)
+from neptune.internal.utils.logger import get_logger
+from neptune.internal.utils.run_state import RunState
+
+if TYPE_CHECKING:
+    from neptune.table import Table
+
+logger = get_logger()
+
+
+def to_pandas(table: Table) -> pd.DataFrame:
+    def make_attribute_value(attribute: AttributeWithProperties) -> Any:
+        _type = attribute.type
+        _properties = attribute.properties
+        if _type == AttributeType.RUN_STATE:
+            return RunState.from_api(_properties.get("value")).value
+        if _type in (
+            AttributeType.FLOAT,
+            AttributeType.INT,
+            AttributeType.BOOL,
+            AttributeType.STRING,
+            AttributeType.DATETIME,
+        ):
+            return _properties.get("value")
+        if _type == AttributeType.FLOAT_SERIES:
+            return _properties.get("last")
+        if _type == AttributeType.STRING_SERIES:
+            return _properties.get("last")
+        if _type == AttributeType.IMAGE_SERIES:
+            return None
+        if _type == AttributeType.FILE or _type == AttributeType.FILE_SET:
+            return None
+        if _type == AttributeType.STRING_SET:
+            return ",".join(_properties.get("values"))
+        if _type == AttributeType.GIT_REF:
+            return _properties.get("commit", {}).get("commitId")
+        if _type == AttributeType.NOTEBOOK_REF:
+            return _properties.get("notebookName")
+        if _type == AttributeType.ARTIFACT:
+            return _properties.get("hash")
+        logger.error(
+            "Attribute type %s not supported in this version, yielding None. Recommended client upgrade.",
+            _type,
+        )
+        return None
+
+    def make_row(entry: LeaderboardEntry) -> Dict[str, Any]:
+        row: Dict[str, Union[str, float, datetime]] = dict()
+        for attr in entry.attributes:
+            value = make_attribute_value(attr)
+            if value is not None:
+                row[attr.path] = value
+        return row
+
+    def sort_key(attr: str) -> Tuple[int, str]:
+        domain = attr.split("/")[0]
+        if domain == "sys":
+            return 0, attr
+        if domain == "monitoring":
+            return 2, attr
+        return 1, attr
```

Comment:
all of the functions `make_attribute_value`, `make_row`, `sort_key` don't share any vars from body of `to_pandas` func. How about moving their definition out of the body of `to_pandas`. You can add `__*` prefix (python convention for private members)

Reply:
On my list already :D Will be done in separate PR.